src/storage/v2/storage.cpp

```diff
@@ -1905,6 +1917,20 @@ utils::BasicResult<Storage::RegisterReplicaError> Storage::RegisterReplica(
   MG_ASSERT(replication_mode == replication::ReplicationMode::SYNC || !config.timeout,
             "Only SYNC mode can have a timeout set");
 
+  if (ShouldStoreAndRestoreReplicas()) {
+    auto data = replication::ReplicaStatusToJSON(
+        replication::ReplicaStatus{.name = name,
+                                   .ip_address = endpoint.address,
+                                   .port = endpoint.port,
+                                   .sync_mode = replication_mode,
+                                   .timeout = config.timeout,
+                                   .replica_check_frequency = config.replica_check_frequency,
+                                   .ssl = config.ssl});
+    if (!kvstorage_->Put(name, data.dump())) {
+      spdlog::error("Issue when saving replica {} in settings.", name);
```

Comment:
Just thinking, not sure, but shouldn't this result in a error reported via the return value? If we couldn't persist the replica, then it means Memgraph cannot pick it up again when it starts. That also mean the error would be turned into an exception in the interpreter which can be easily seen by the user.